app/helpers/payouts_helper.rb

```diff
@@ -102,18 +170,64 @@ def old_payout_period_data(user:, payment:)
 
     payout_period_data[:payout_date_formatted] = formatted_payout_date(payment.created_at)
     payout_period_data[:payout_currency] = payment.currency
-    payout_period_data[:payout_cents] = payment.amount_cents
-    payout_period_data[:payout_displayed_amount] = payment.displayed_amount
+    payout_total_cents = payment.balances.sum(:holding_amount_cents)
+    payout_period_data[:payout_cents] = payout_total_cents
+    payout_period_data[:payout_displayed_amount] = formatted_dollar_amount(payout_total_cents)
     payout_period_data[:is_processing] = payment.processing?
     payout_period_data[:arrival_date] = payment.arrival_date ? formatted_payout_date(Time.zone.at(payment.arrival_date)) : nil
     payout_period_data[:status] = payment.state
+    if payment.failed?
+      payout_period_data[:displayable_failure_reason] = payment.displayable_failure_reason
+      payout_period_data[:requires_verification_to_resume] = payment.requires_verification_to_resume?
+    end
     payout_period_data[:payment_external_id] = payment.external_id
     payout_period_data[:type] = payment.payout_type || Payouts::PAYOUT_TYPE_STANDARD
 
     payout_period_data[:payout_note] = nil
 
     balance_ids = payment.balances.map(&:id)
-    payout_period_data.merge!(payout_sales_data(user:, balance_ids:,
+    carried_over_balance_ids = []
+
+    if balance_ids.present? && payment.completed?
+      earlier_failed_list = user.payments
+        .failed
+        .joins(:balances)
+        .where("payments.created_at < ?", payment.created_at)
+        .where(balances: { id: balance_ids })
+        .order("payments.created_at ASC")
+        .distinct
+
+      if earlier_failed_list.exists?
```

Comment:
It seems this entire logic is duplicated from above, can we please move to a common method and use at both places?